### ISSUE TITLE
diff_sph.py example doesn't work when training the model yourself

### ISSUE THREAD

**Issue by gpu-daal** [taichi-dev/taichi#8632]
Training the model yourself with default parameters provides the output that doesn't even remotely work. It's almost a constant stream of particles with constant direction and force, despite the final loss value being ~0.18 (it doesn't converge beyond that no matter how many epochs you run). It doesn't try to hit the box as the pretrained model does.
BTW, pretrained model gives around 0.5 loss when being trained without backward pass.

How to reproduce:
python diff_sph.py --train
*change the code to make the program consume trained weights and biases*
python diff_sph.py
